manager/state/raft/raft.go

```diff
@@ -1159,3 +1178,16 @@ func getIDs(snap *raftpb.Snapshot, ents []raftpb.Entry) []uint64 {
 	}
 	return sids
 }
+
+// finalize cancels the context of all the pending proposals on raft node
+// shutdown. This is used not to block the manager indefinitely while the
+// quorum was lost. This should be called only after all the components
+// using the raft store have been shut down, not submitting any more
+// proposals. This can lead to undeterminate behavior or deadlock otherwise.
+func (n *Node) finalize() {
+	n.proposeLock.Lock()
+	defer n.proposeLock.Unlock()
+	for _, cancel := range n.proposals {
+		cancel()
+	}
```

Comment:
I think we should have a waitgroup for proposals and wait here for the cancellations to take effect. It would be bad to destroy the raft node while there are still active proposals.